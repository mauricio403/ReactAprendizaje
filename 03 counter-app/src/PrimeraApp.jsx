import React  from 'react'
import PropTypes from 'prop-types'
//import React, { Fragment } from 'react'


const PrimeraApp = ({ saludo, subtitulo  })=>{
    
    // const saludo = 'Hola mundo ';

     //console.log(props)




    return (
        <>
            <h1>{saludo}</h1>
            {/* <pre>{ JSON.stringify(saludo, null,3) }</pre> */}
            <p>{ subtitulo }</p>
        </>
        
    );
    
}

//Aqui se define una propiedad obligatoria que tiene el componente
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

//Aqui defino una propiedad del componente por defecto
PrimeraApp.defaultProps ={
    subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;