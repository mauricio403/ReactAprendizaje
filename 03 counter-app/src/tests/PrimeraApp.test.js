
import PrimeraApp from "../PrimeraApp";
import React from 'react';
import {shallow} from 'enzyme'
describe('Pruebas en primera app', () => {
    // test('Debe mostrar el mensaje "Hola, soy goku" ', () => {
        
    //     const saludo = 'Hola soy goku';
    //     const {getByText} = render( <PrimeraApp saludo={saludo}/> )

    //   expect(getByText(saludo)).toBeInTheDocument();
    // })
    
  test('Debe mmostrar la PrimeraPp correctamente ', () => {
    

    const saludo = 'Hola soy goku'

    const wrapper = shallow(<PrimeraApp saludo={saludo}/>);

    expect(wrapper).toMatchSnapshot();

  })
  test('Deme mostrar el subtitulo de props ', () => {
    
    const saludo = 'Hola soy goku';
    const subtitulo = 'Dame tu fuerza pegaso'

    const wrapper = shallow(
    <PrimeraApp 
    saludo={saludo} 
    subtitulo = { subtitulo }

  
    />);

      const textoParrafo = wrapper.find('p').text();
    
      expect(textoParrafo).toBe(subtitulo)

  })
  
  

})
