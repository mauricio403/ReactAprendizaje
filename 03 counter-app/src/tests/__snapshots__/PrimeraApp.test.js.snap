// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en primera app Debe mmostrar la PrimeraPp correctamente  1`] = `
<Fragment>
  <h1>
    Hola soy goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
